src/OpenTelemetry.AutoInstrumentation.Loader/AppConfigUpdater.cs

```diff
@@ -0,0 +1,28 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+#if NETFRAMEWORK
+using System.Text;
+using System.Xml;
+using System.Xml.Linq;
+
+namespace OpenTelemetry.AutoInstrumentation.Loader;
+
+#pragma warning disable RS0016
+#pragma warning restore RS0016
+
+/// <summary>
+/// Handles update of config files for non-default AppDomain
+/// </summary>
+public static class AppConfigUpdater
+{
+    /// <summary>
+    /// Modify assembly bindings in appDomainSetup
+    /// </summary>
+    /// <param name="appDomainSetup">appDomainSetup to be updated</param>
+    public static void ModifyConfig(AppDomainSetup appDomainSetup)
+    {
+        appDomainSetup.LoaderOptimization = LoaderOptimization.SingleDomain;
```

Comment:
We always force application domain to be `LoaderOptimization.SingleDomain`, as it is `LoaderOptimization.MultiDomainHost` does not work. It will be a good idea to check, if `LoaderOptimization.MultiDomain` works and not reset it in that case.
Further work would be done in phase 2, where LoaderOptimization would be changed only for non-GAC case by default.
We probably should add in phase 2/3 environment variable that would allow disable that modification, as `LoaderOptimization.SingleDomain` will work in most cases after phase 2.